src/services/geometry/richgeo/ReadoutGeo.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2023, Christopher Dilks, Luigi Dello Stritto
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+//
+
+#include "ReadoutGeo.h"
+
+// constructor
+richgeo::ReadoutGeo::ReadoutGeo(std::string detName_, dd4hep::Detector *det_, std::shared_ptr<spdlog::logger> log_)
+  : m_detName(detName_), m_det(det_), m_log(log_)
+{
+  // capitalize m_detName
+  std::transform(m_detName.begin(), m_detName.end(), m_detName.begin(), ::toupper);
+
+  // random number generators
+  m_random.SetSeed(1); // default seed
+
+  // default (empty) cellID looper
+  m_loopCellIDs = [] (std::function<void(uint64_t)> lambda) { return; };
+
+  // default (empty) cellID rng generator
+  m_rngCellIDs = [] (std::function<void(uint64_t)> lambda, float p) { return; };
+
+  // common objects
+  m_readoutCoder = m_det->readout(m_detName+"Hits").idSpec().decoder();
+  m_detRich      = m_det->detector(m_detName);
+  m_systemID     = m_detRich.id();
+
+  // dRICH readout --------------------------------------------------------------------
+  if(m_detName=="DRICH") {
+
+    // get number of sectors
+    m_num_sec = m_det->constant<int>("DRICH_num_sectors");
+    // get number of sensors
+    m_num_mod = m_det->constant<int>("DRICH_num_sensors");
+    // get number of pixels along one side of a SiPM
+    m_num_px = m_det->constant<int>("DRICH_num_px");
+
+    // define cellID looper
+    m_loopCellIDs = [this] (std::function<void(uint64_t)> lambda) {
```

Comment:
Not clear why this can't be a method...

Reply:
1. You don't explain what kind of flexibility
2. Could it be that the code simply doesn't belong in the service?

Reply:
> Could it be that the code simply doesn't belong in the service?

Where do you think it should go?

Reply:
> 1. You don't explain what kind of flexibility

It will be more clear what we need when the pfRICH (or others) starts using these algorithms.

Reply:
> > Could it be that the code simply doesn't belong in the service?
> 
> Where do you think it should go?

In the algorithm itself.

Reply:
This function is specific to the dRICH, and no part of `PhotoMultiplierHitDigi` should be subsystem-specific. `richgeo` is designed to keep PID-subsystem-dependent code out of all algorithms.